homeassistant/components/axis/binary_sensor.py

```diff
@@ -23,26 +26,159 @@
 from .entity import AxisEventEntity
 from .hub import AxisHub
 
-DEVICE_CLASS = {
-    EventGroup.INPUT: BinarySensorDeviceClass.CONNECTIVITY,
-    EventGroup.LIGHT: BinarySensorDeviceClass.LIGHT,
-    EventGroup.MOTION: BinarySensorDeviceClass.MOTION,
-    EventGroup.SOUND: BinarySensorDeviceClass.SOUND,
-}
-
-EVENT_TOPICS = (
-    EventTopic.DAY_NIGHT_VISION,
-    EventTopic.FENCE_GUARD,
-    EventTopic.LOITERING_GUARD,
-    EventTopic.MOTION_DETECTION,
-    EventTopic.MOTION_DETECTION_3,
-    EventTopic.MOTION_DETECTION_4,
-    EventTopic.MOTION_GUARD,
-    EventTopic.OBJECT_ANALYTICS,
-    EventTopic.PIR,
-    EventTopic.PORT_INPUT,
-    EventTopic.PORT_SUPERVISED_INPUT,
-    EventTopic.SOUND_TRIGGER_LEVEL,
+
+@dataclass(frozen=True, kw_only=True)
+class AxisBinarySensorDescription(BinarySensorEntityDescription):
+    """Axis binary sensor entity description."""
+
+    event_topic: tuple[EventTopic, ...] | EventTopic
+    """Event topic that provides state updates."""
+    name_fn: Callable[[AxisHub, Event], str] = lambda hub, event: ""
+    """Function providing the corresponding name to the event ID."""
+    supported_fn: Callable[[AxisHub, Event], bool] = lambda hub, event: True
+    """Function validating if event is supported."""
+
+
+@callback
+def event_id_is_int(event_id: str) -> bool:
+    """Make sure event ID is int."""
+    try:
+        return isinstance(int(event_id), int)
```

Comment:
We can avoid the use of `isinstance` here (which is quite expensive)?

```suggestion
        int(event_id)
        return True
```

Reply:
Fixed